### PR TITLE
Update `PatientStatus` to migrated

### PR DESCRIPTION
**Pull request by Honey14**
https://app.clubhouse.io/simpledotorg/story/2359/update-patient-migrated-status-from-call-result

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/contactpatient/ContactPatientEffect.kt [hunk 1]
```diff
@@ -47,3 +47,7 @@ data class CancelAppointment(
 data class MarkPatientAsMovedToPrivate(
     val patientUuid : UUID
 ): ContactPatientEffect()
+
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/contactpatient/ContactPatientEffect.kt:50]: Same feedback as earlier about a non-descriptive commit message. Please reword?

File: app/src/main/java/org/simple/clinic/contactpatient/ContactPatientEffect.kt [hunk 2]
```diff
@@ -47,3 +47,7 @@ data class CancelAppointment(
...
+data class RemoveAppointment(
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/contactpatient/ContactPatientEffect.kt:51]: We already have a `CancelAppointment` effect. Do we need a new effect here, when that one accomplishes the same task?

File: app/src/main/java/org/simple/clinic/contactpatient/ContactPatientEffectHandler.kt [hunk 3]
```diff
@@ -46,9 +46,19 @@ class ContactPatientEffectHandler @AssistedInject constructor(
...
+        .addTransformer(MarkPatientAsTransferredToAnotherFacility::class.java, markPatientAsMovedToPublic(schedulers.io()))
         .build()
   }
 
+  private fun markPatientAsMovedToPublic(scheduler: Scheduler): ObservableTransformer<MarkPatientAsTransferredToAnotherFacility, ContactPatientEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(scheduler)
+          .doOnNext { patientRepository.updatePatientStatusToMigrated(patientUuid = it.patientUuid) }
+          .map { PatientMarkAsMigrated }
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/contactpatient/ContactPatientEffectHandler.kt:58]: The `RemoveAppointment` effect which is triggered by this event will end up hard-coding the appointment cancel reason to `MovedToPrivatePractioner` which is the wrong reason here.

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/simple/clinic/contactpatient/ContactPatientEffectHandler.kt (hunk 3 above)
Comment (Honey14) [reply app/src/main/java/org/simple/clinic/contactpatient/ContactPatientEffectHandler.kt:58]: Got it. Removing the `RemoveAppointment` effect and continue using the `CancelAppointment` effect

**Review by vinaysshenoy — APPROVED** [review]

**Pull request merged**